src/daemon/rpmostree-sysroot-upgrader.c

```diff
@@ -1189,19 +1105,18 @@ overlay_packages (RpmOstreeSysrootUpgrader *self,
   else
     base_rev = g_strdup (self->new_revision);
 
-  /* create a tmprootfs in which we initially only check out the bare minimum to
-   * make libhif happy. this allows us to provide more immediate feedback to the
-   * user if e.g. resolving/downloading/etc... goes wrong without making them
-   * wait for a full tree checkout first. */
-  if (!checkout_min_tree_in_tmp (repo, base_rev, &tmprootfs, &tmprootfs_dfd,
-                                 cancellable, error))
-    goto out;
+  if (g_strv_length (self->requested_packages) > 0)
```

Comment:
Note that I also kept the optimization that we don't do the temporary checkout if we don't have layered packages.